addons/wowl/static/src/services/ui/ui.ts

```diff
@@ -0,0 +1,106 @@
+import { Component, core, tags, useState } from "@odoo/owl";
+import type { OdooEnv, Service } from "../../types";
+
+const { EventBus } = core;
+
+export interface UIService {
+  block: () => void;
+  unblock: () => void;
+}
+
+class BlockUI extends Component<{}, OdooEnv> {
+  static template = tags.xml`
+    <div>
+      <div t-if="state.blockUI" class="o_blockUI" >
+        <div class="o_spinner">
+            <img src="/wowl/static/img/spin.png" alt="Loading..."/>
+        </div>
+        <div class="o_message">
+            <t t-raw="state.message"/>
+        </div>
+      </div>
+    </div>`;
+  messagesByDuration = [
+    { time: 2, text: this.env._t("Loading...") },
+    { time: 4, text: this.env._t("Still loading...") },
+    { time: 6, text: this.env._t("Still loading...<br />Please be patient.") },
+    { time: 180, text: this.env._t("Don't leave yet,<br />it's still loading...") },
+    {
+      time: 120,
+      text: this.env._t("You may not believe it,but the application is actually loading..."),
+    },
+    {
+      time: 3180,
+      text: this.env._t("Take a minute to get a coffee,<br />because it's loading..."),
+    },
+    {
+      time: null,
+      text: this.env._t("Maybe you should consider reloading the application by pressing F5..."),
+    },
+  ];
+  state = useState({
+    blockUI: false,
+    message: "",
+    count: 0,
+  });
+  msgTimer!: number;
+
+  replaceMessage(idx: number) {
+    const message = this.messagesByDuration[idx];
+    this.state.message = message.text;
+    if (message.time !== null) {
+      this.msgTimer = setTimeout(() => {
+        this.replaceMessage(idx + 1);
+      }, message.time * 1000);
+    }
+  }
+
+  block() {
+    if (this.state.count === 0) {
+      this.state.blockUI = true;
+      this.replaceMessage(0);
+    }
+    this.state.count++;
+  }
+
+  unblock() {
+    if (this.state.count > 0) {
+      this.state.count--;
+    }
+    if (this.state.count === 0) {
+      this.state.blockUI = false;
+      clearTimeout(this.msgTimer);
+      this.state.message = "";
+    }
+  }
+}
+
+export const uiService: Service<UIService> = {
+  name: "ui",
+  deploy(env: OdooEnv): UIService {
+    const bus = new EventBus();
+
+    class ReactiveBlockUI extends BlockUI {
+      constructor() {
+        super(...arguments);
+        bus.on("BLOCK", this, () => {
+          this.block();
+        });
+        bus.on("UNBLOCK", this, () => {
+          this.unblock();
+        });
```

Comment:
Maybe this would work:
```suggestion
        bus.on("BLOCK", this, this.block);
        bus.on("UNBLOCK", this, this.unblock);
```